tests-integration/tests/ibc_registry.rs

```diff
@@ -0,0 +1,144 @@
+use andromeda_app::app::{AppComponent, ComponentType};
+use andromeda_app_contract::mock::{mock_andromeda_app, MockAppContract};
+use andromeda_ibc_registry::mock::{
+    mock_andromeda_ibc_registry, mock_ibc_registry_instantiate_msg, MockIbcRegistry,
+};
+use andromeda_std::{
+    amp::AndrAddr,
+    error::ContractError,
+    os::ibc_registry::{AllDenomInfoResponse, DenomInfo, DenomInfoResponse, IBCDenomInfo},
+};
+use andromeda_testing::{mock::mock_app, mock_builder::MockAndromedaBuilder, MockContract};
+use cosmwasm_std::coin;
+
+#[test]
+fn test_ibc_registry() {
+    let mut router = mock_app(None);
+    let andr = MockAndromedaBuilder::new(&mut router, "admin")
+        .with_wallets(vec![
+            ("owner", vec![coin(100_000, "uandr"), coin(100_000, "uusd")]),
+            ("service_address", vec![]),
+            ("user1", vec![]),
+        ])
+        .with_contracts(vec![
+            ("app-contract", mock_andromeda_app()),
+            ("ibc_registry", mock_andromeda_ibc_registry()),
+        ])
+        .build(&mut router);
+    let owner = andr.get_wallet("owner");
+    let service_address = andr.get_wallet("service_address");
+    let user1 = andr.get_wallet("user1");
+
+    let app_code_id = andr.get_code_id(&mut router, "app-contract");
```

Comment:
The plan is to have this be an OS module under the key `ibc-registry`, we should add it as such for more realistic tests.